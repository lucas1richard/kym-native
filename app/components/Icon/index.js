/**
*
* Icon
*
*/

import React from 'react';
import PropTypes from 'prop-types';
// import { View } from 'react-native';
import iftheme from 'theme/iftheme';
import SvgIcon from './SvgIcon';

import icons from './icons';

const Icon = ({ name, fill, style, ...props }) => {
  const thmFill = iftheme[fill] || iftheme.green;
  // return (<View />);
  return (<SvgIcon {...props} name={name} fill={thmFill} {...props} svgs={icons} />);
};

export default Icon;

Icon.propTypes = {
  fill: PropTypes.string,
  name: PropTypes.string.isRequired,
  props: PropTypes.object,
  className: PropTypes.string,
  style: PropTypes.object,
  onClick: PropTypes.func
};

Icon.defaultProps = {
  fill: iftheme.green,
  viewBox: '0 0 50 50'
};
