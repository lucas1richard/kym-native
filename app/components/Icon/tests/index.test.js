import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import React from 'react';

import Icon from '../index';

const renderedComponent = (props = {}) => mount(<Icon {...props} />);

describe('<Icon />', () => {
  it('should render an svg', () => {
    expect(renderedComponent({ name: 'leftArrow' }).find('svg').length).toEqual(1);
  });

  it('should accept a name prop', () => {
    expect(renderedComponent({ name: 'leftArrow' }).prop('name')).toEqual('leftArrow');
  });

  it('should call handleClick function once', () => {
    const handleClick = jest.fn();
    renderedComponent({ onClick: handleClick, name: 'leftARrow' }).find('svg').simulate('click');
    expect(handleClick.mock.calls.length).toBe(1);
  });

  it('mounts my component properly', () => {
    const props = {
      width: '25',
      name: 'leftArrow',
      onClick: jest.fn()
    };
    expect(mountToJson(renderedComponent(props))).toMatchSnapshot();
  });
});
