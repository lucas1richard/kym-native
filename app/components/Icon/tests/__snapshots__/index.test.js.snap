// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> mounts my component properly 1`] = `
<Icon
  name="leftArrow"
  onClick={[Function]}
  width="25"
>
  <styled.svg
    onClick={[Function]}
    viewBox="0 0 50 50"
    width="25"
    xmlns="http://www.w3.org/2000/svg"
  >
    <svg
      className="sc-bdVaJa dhcYXN"
      onClick={[Function]}
      viewBox="0 0 50 50"
      width="25"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M 19.8125 13.09375 A 1.0001 1.0001 0 0 0 19.25 13.40625 L 8.34375 24.28125 L 7.65625 25 L 8.34375 25.71875 L 19.25 36.59375 A 1.0001 1.0001 0 1 0 20.65625 35.1875 L 11.46875 26 L 41 26 A 1.0001 1.0001 0 1 0 41 24 L 11.46875 24 L 20.65625 14.8125 A 1.0001 1.0001 0 0 0 19.8125 13.09375 z"
      />
    </svg>
  </styled.svg>
</Icon>
`;
